### DISCUSSION TITLE
how to get the file content in subdirectory？

### DISCUSSION THREAD

**Discussion by supler** [python-gitlab/python-gitlab#2650] [Q&A] [answered]
it will return 404 error using the below codes

    with gitlab.Gitlab(url=GITLAB_URL, private_token=TOKEN) as gl:
        p = gl.projects.get(4)
        pom_file = p.files.get(file_path='lal-parent/pom.xml', ref=branch_name)

git likes as the below

lal-imp/
 xx
 lal-parent/pom.xml  --------this is the file I wanted
 pom.xml

**Comment by nejch** [comment, marked as answer]
@supler this should actually work. So this file either does not exist on the branch you're looking for, or you have some other issue like potentially fetching the wrong project, file path, branch etc.

**Reply by supler** [reply]
but I always get error as the below

gitlab.exceptions.GitlabListError: 404: 404 Not Found

**Reply by metaloid2k21** [reply]
same exact issue here :( still didn't find a plausible solution

**Reply by JohnVillalovos** [reply]
Will need more details.

Also you might try doing:
`gl.enable_debug()` to see what is being sent to the GitLab server.

Also helpful to look at the upstream docs:
https://docs.gitlab.com/ee/api/repository_files.html#get-file-from-repository

**Reply by metaloid2k21** [reply]
After activating the debug mode on the GitLab object instance, I was able to identify the the wrongly passed parameters.
It's now working as expected.
Tks for the hint 👍

**Reply by JohnVillalovos** [reply]
> After activating the debug mode on the GitLab object instance, I was able to identify the the wrongly passed parameters. It's now working as expected. Tks for the hint 👍

Great to hear!
